x-pack/plugin/relevance-search/src/main/java/org/elasticsearch/xpack/relevancesearch/settings/AbstractSettingsService.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.relevancesearch.settings;
+
+import org.elasticsearch.client.internal.Client;
+import org.elasticsearch.index.IndexNotFoundException;
+import org.elasticsearch.xpack.relevancesearch.settings.index.IndexCreationService;
+
+import java.util.Map;
+
+public abstract class AbstractSettingsService<S extends Settings> {
```

Comment:
Maybe just `SettingsService`, or `BaseSettingsService`? It seems redundant to put `Abstract` in a name.

Reply:
Prepending abstract classes with `Abstract` seems like a convention in the Elasticsearch codebase.

Reply:
Yep, sorry about that... but it's a convention in several codebases. Similar to naming `Base` in Ruby for base classes. 🤷